spec/controllers/assignments_controller_spec.rb

```diff
@@ -90,25 +95,30 @@
 
   describe 'POST #generate_rotation' do
     before :each do
+      @roster = create :roster
       @start_date = Date.today.strftime '%Y-%m-%d'
       @end_date = Date.tomorrow.strftime '%Y-%m-%d'
-      user_1 = create :user
-      user_2 = create :user
-      user_3 = create :user
+      user_1 = create :user, rosters: [@roster]
+      user_2 = create :user, rosters: [@roster]
+      user_3 = create :user, rosters: [@roster]
       @user_ids = [user_1.id.to_s, user_2.id.to_s, user_3.id.to_s]
       @starting_user_id = @user_ids[1]
       when_current_user_is :whoever
     end
     let :submit do
       post :generate_rotation,
+           roster_id: @roster.id,
            start_date: @start_date,
            end_date: @end_date,
            user_ids: @user_ids,
            starting_user_id: @starting_user_id
     end
     it 'calls Assignment#generate rotation with the given arguments' do
-      expect(Assignment).to receive(:generate_rotation)
-        .with @user_ids, Date.today, Date.tomorrow, @starting_user_id
+      # remove all other instances of roster so 'any instance' definitely
+      # refers to our @roster instance
+      Roster.where.not(id: @roster.id).delete_all
```

Comment:
when_current_user_is creates a user. Creating a user creates a roster, since there can never be any users unless there are rosters.